pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -222,24 +241,19 @@ func (c *DeploymentConfigController) findDetails(config *deployapi.DeploymentCon
 		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
 			if !errors.IsNotFound(err) {
 				glog.V(2).Infof("Error while trying to get image stream tag %q: %v", istag, err)
-				return details, existingDeployments, latestDeploymentExists, nil
+				return details, nil
 			}
 			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
 			// (a build configuration output points to it so it's going to be populated at some point in the
 			// future)
 			details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream tag %q does not exist.", istag, istag)
-			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
-			if err != nil {
-				glog.V(2).Infof("Error while trying to list build configs: %v", err)
-				return details, existingDeployments, latestDeploymentExists, nil
-			}
-
-			for _, bc := range bcList.Items {
+			for _, obj := range c.buildConfigs.List() {
```

Comment:
I don't like this - this is a best effort error check, and looking at all build configs (and holding them all in memory) is expensive for the value we get.

I would only look at build configs in the current namespace at most.  I don't like that we have to do an expensive list here in order to report an error, but if this code change *guarantees* that we don't do this too often (no more than a minute or when it is changed) I could maybe live with it.